sklearn/metrics/pairwise.py

```diff
@@ -1216,6 +1369,14 @@ def pairwise_distances(X, Y=None, metric="euclidean", n_jobs=1, **kwds):
                          "Valid metrics are %s, or 'precomputed', or a "
                          "callable" % (metric, _VALID_METRICS))
 
+    if metric in _MASKED_METRICS or callable(metric):
+        missing_values = kwds.get("missing_values") if kwds.get(
+            "missing_values") is not None else np.nan
+
+        if np.all(_get_mask(X, missing_values)):
```

Comment:
maybe ``_get_missing_mask``? It took me a double take to understand it because I kind of assumed the function would return the non-missing mask at first...